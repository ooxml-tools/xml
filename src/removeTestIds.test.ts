import { describe, expect, test } from "vitest";
import { removeTestIds } from "./";
import { js2xml, xml2js } from "xml-js";

describe("removeTestIds", () => {
  test("removed all _testid attributes", () => {
    const xml = `
      <root>
        <foo _testid="foobar">
          <foo _testid="foobar">
            TEST
          </foo>
        </foo>
        <bar _testid="foobar">
          TEST
        </bar>
      </root>
    `;
    expect(xml).toMatch(/_testid/);
    const root = xml2js(xml);
    const output = removeTestIds(root);
    const outputXml = js2xml(output)
    expect(outputXml).not.toMatch(/_testid/);
    expect(output).toMatchSnapshot();
  });
});
