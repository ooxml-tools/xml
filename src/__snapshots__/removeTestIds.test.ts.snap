// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`removeTestIds > removed all _testid attributes 1`] = `
{
  "attributes": undefined,
  "elements": [
    {
      "attributes": undefined,
      "elements": [
        {
          "attributes": {},
          "elements": [
            {
              "attributes": {},
              "elements": [
                {
                  "attributes": undefined,
                  "elements": undefined,
                  "text": "
            TEST
          ",
                  "type": "text",
                },
              ],
              "name": "foo",
              "type": "element",
            },
          ],
          "name": "foo",
          "type": "element",
        },
        {
          "attributes": {},
          "elements": [
            {
              "attributes": undefined,
              "elements": undefined,
              "text": "
          TEST
        ",
              "type": "text",
            },
          ],
          "name": "bar",
          "type": "element",
        },
      ],
      "name": "root",
      "type": "element",
    },
  ],
}
`;
